mops/src/opsa/cpu.tpp

```diff
@@ -1,102 +1,96 @@
-#include <cassert>
 #include <algorithm>
+#include <cassert>
 #include <stdexcept>
 #include <string>
 
 #include "mops/opsa.hpp"
 
 using namespace mops;
 
-template<typename scalar_t>
-static void check_inputs_shape(
-    Tensor<scalar_t, 2> A,
-    Tensor<scalar_t, 2> B,
-    Tensor<int32_t, 1> indices_output
-) {
+template <typename scalar_t>
+static void check_inputs_shape(Tensor<scalar_t, 2> A, Tensor<scalar_t, 2> B,
+                               Tensor<int32_t, 1> indices_output) {
     if (A.shape[0] != B.shape[0]) {
         throw std::runtime_error(
             "A and B tensors must have the same number of elements along the "
-            "first dimension, got " + std::to_string(A.shape[0]) + " and " +
-            std::to_string(B.shape[0])
-        );
+            "first dimension, got " +
+            std::to_string(A.shape[0]) + " and " + std::to_string(B.shape[0]));
```

Comment:
You might be able to re-use the checks that we have in main now. I've tried to abstract all of them out into a few functions and you might be able to copy and paste most of them from the CPU code (except those that do index bound checks)